### PR TITLE
feat(waitFor): improve error message for non-function callbacks

### LINKED ISSUE #509: Improve error message for non-function arg passed to `waitFor`
### Describe the feature you'd like:
New `waitFor` async utils expects a callback function as first argument. Though if something else is passed there, the error looks like this:
wait-for

It's difficult to figure out the actual error with that message (it took me few minutes to found this in a project where I was migrating dom-testing-library to the last version!). I'd like to get a better message for this case.

### Suggested implementation:
This would be as simple as throw an error at first line of `waitFor` implementation if `typeof callback !== 'function'`. I'm happy to implement this myself, though I'm not sure which message should I write so I'd like to discuss that here.

### Describe alternatives you've considered:
n/a

### Teachability, Documentation, Adoption, Migration Strategy:
n/a

### PR DESCRIPTION
**Pull request by Belco90**
**What**:
Closes #509

**Why**:
To improve error message and error stack trace

**How**:
Checking if `callback` is a function.

**Checklist**:

- [ ] Documentation added to the
      [docs site](https://github.com/testing-library/testing-library-docs)
- [ ] I've prepared a PR for types targeting
      [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/testing-library__dom)
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

File: src/wait-for.js [hunk 1]
```diff
@@ -22,6 +22,10 @@ function waitFor(
     },
   } = {},
 ) {
+  if (typeof callback !== 'function') {
+    throw new Error('Received `callback` arg must be a function')
```
Comment (eps1lon) [inline src/wait-for.js]:
```suggestion
    throw new TypeError('Received `callback` arg must be a function')
```

File: src/__tests__/wait-for.js [hunk 2]
```diff
@@ -32,3 +32,9 @@ test('uses generic error if there was no last error', async () => {
   ).catch(e => e)
   expect(result).toMatchInlineSnapshot(`[Error: Timed out in waitFor.]`)
 })
+
+test('throws nice error if provided callback is not a function', () => {
+  expect(() => waitFor('not a function')).toThrow(
```
Comment (eps1lon) [inline src/__tests__/wait-for.js]: Could you use a more reasonable argument here like a DOM element?

Comment (Belco90) [reply src/__tests__/wait-for.js]: Done. I hope that element I'm passing looks fine.

**Comment by eps1lon** [comment]
@Belco90 Looking good, thanks!

**Pull request merged**